cpp/src/arrow/array.cc

```diff
@@ -48,6 +48,14 @@ bool Array::EqualsExact(const Array& other) const {
   return true;
 }
 
+bool Array::RangeEqualsExact(int32_t start_idx, int32_t end_idx, const Array& arr) const {
+  if (this == &arr) { return true; }
+  for (int i = start_idx; i < end_idx; ++i) {
```

Comment:
Are you thinking of the case of start_idx == end_idx or something else?  

All else being equal it seems that an empty set should equal an empty.  Do you disagree?